src/mp_units.cpp

```diff
@@ -0,0 +1,74 @@
+// The MIT License (MIT)
```

Comment:
It seems that this is no longer a header-only library, right? It will produce different artifacts for different compilers. So probably we should get rid of the: https://github.com/mpusz/units/blob/70586138d9590076a3b46b12133956cf97dcfc41/conanfile.py#L145-L146 in case modules are being used (additional Conan option needed).

Reply:
Just not "only". The header-only library targets are still there, and are even used by `mp-units::module` through its linked library `mp-units::systems`.

Reply:
Yes, but as long as there is at least one artifact that is compiler-dependent, Conan has to generate a different ID for such configurations. This is why we need a module-specific ON/OFF option in Conan and when it is on we should not use `self.info.header_only()` as this forces generation of the same identifier for the package (no matter what the compiler settings are).

Reply:
I see. Do I just need to remove those lines?

Reply:
Yes, add a new Conan option and make a branch that will run it only when modules are not being used.

Reply:
Also please add docs here: https://mpusz.github.io/units/usage.html#conan-options for both Conan and CMake.